### PR TITLE
Spec: typo/layout fixes

### PR DESCRIPTION
**Pull request by Kegsay**
This PR fixes a number of typos/spelling errors/layout bugs/TODO comments which weren't hidden from the generated output. It also begins to split up unrelated sections into their own files (e.g. identity server from appendices)

Finally, it fixes a bug with event schemas introduced in https://github.com/matrix-org/matrix-doc/pull/48 which rendered the kind of event as `None` instead of `Message Event` `State Event` etc.

None of this should be controversial.

### CODE REVIEW

File: specification/00_basis.rst [hunk 1]
```diff
@@ -442,7 +444,7 @@ The ``error`` string will be a human-readable error message, usually a sentence
 explaining what went wrong. The ``errcode`` string will be a unique string
 which can be used to handle an error message e.g. ``M_FORBIDDEN``. These error
 codes should have their namespace first in ALL CAPS, followed by a single _ to
-ease seperating the namespace from the error code.. For example, if there was a
+ease separating the namespace from the error code.. For example, if there was a
```
Comment (illicitonion) [inline specification/00_basis.rst]: Drop the second . in .. ?

File: specification/45_typing_notifications.rst [hunk 2]
```diff
@@ -5,9 +5,10 @@ Client APIs
...
 
 To set "I am typing for the next N msec"::
+
   PUT .../rooms/<room_id>/typing/<user_id>
   Content:  { "typing": true, "timeout": N }
-  # timeout is in msec; I suggest no more than 20 or 30 seconds
+  # timeout is in milliseconds; suggested no more than 20 or 30 seconds
 
 This should be re-sent by the client to continue informing the server the user
 is still typing; I suggest a safety margin of 5 seconds before the expected
```
Comment (illicitonion) [inline specification/45_typing_notifications.rst]: "I suggest" is somewhat unusual phrasing for a specification.

**Comment by illicitonion** [comment]
Minor nits, otherwise LGTM

File: specification/00_basis.rst (hunk 1 above)
Comment (Kegsay) [inline specification/00_basis.rst]: Done.

File: specification/45_typing_notifications.rst (hunk 2 above)
Comment (Kegsay) [inline specification/45_typing_notifications.rst]: Done.

**Pull request merged**